helm/cluster/templates/clusterapi/workers/machinepool.yaml

```diff
@@ -27,14 +27,14 @@ metadata:
   namespace: {{ $.Release.Namespace }}
 spec:
   clusterName: {{ include "cluster.resource.name" $ }}
-  replicas: {{ $nodePoolConfig.replicas }}
+  replicas: {{ $nodePoolConfig.minSize | default $nodePoolConfig.replicas }}
   template:
     spec:
       bootstrap:
         configRef:
           apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
           kind: KubeadmConfig
-          name: {{ include "cluster.resource.name" $ }}-{{ $nodePoolName }}
+          name: {{ include "cluster.resource.name" $ }}-{{ $nodePoolName }}-{{ include "cluster.internal.workers.kubeadmConfig.spec.hash" $ }}
```

Comment:
Why do we need to add an hash here?

Reply:
TL;DR would be that there is bug where AWSMachinePool nodes do not do rolling update when referenced `KubeadmConfig` changes, so here we append the `KubeadmConfig` hash to the CR name, so when KubeadmConfig spec changes the CR name changes as well, and AWSMachinePool gets a new KubeadmConfig, which correctly triggers a rolling update. (we already do something similar for the control plane as well)

I have ported [this change](https://github.com/giantswarm/cluster-aws/pull/457) from cluster-aws, the explanation (copied from cluster-aws as well) with issue links is here in the PR https://github.com/giantswarm/cluster/pull/70/files#diff-bb8c36a7911fe46567a52f4b7ba28fe0bba14a08572f2250180eb1917a800454R2

Reply:
Oh wow, thank you.